### PR TITLE
Support named arguments when resolving template parameters

### PR DESCRIPTION
**Pull request by rvanvelzen**
Resolves phpstan/phpstan#5262

### CODE REVIEW

**Comment by rvanvelzen** [comment]
Perhaps this should target 1.5.x?

File: tests/PHPStan/Analyser/NodeScopeResolverTest.php [hunk 1]
```diff
@@ -848,6 +848,12 @@ public function dataFileAsserts(): iterable
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-3853.php');
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-6936-limit.php');
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-6927.php');
+
+		if (PHP_VERSION_ID < 80000) {
+			return;
+		}
+
+		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-5262.php');
```
Comment (staabm) [inline tests/PHPStan/Analyser/NodeScopeResolverTest.php]: the cs-fixer is doing this ugly thing a lot, and I don't like it.

```suggestion
		if (PHP_VERSION_ID >= 80000) {
    		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-5262.php');
		}

		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-6927.php');
```

to work arround such silly IF which might lead to the next person to append a new test-case which then accidentally only runs on php < 8, I am using a different construction

Comment (rvanvelzen) [reply tests/PHPStan/Analyser/NodeScopeResolverTest.php]: Yeah, I didn't like it either but forgot to do something about it. Thanks!

**Comment by staabm** [comment]
just to make sure it isn't forgotten: regarding named arguments there is some other work prepared in https://github.com/phpstan/phpstan-src/pull/750 , where we also had some discussion.

this PR touches other files though, so it seems there is no overlap and everything is fine.

**Comment by ondrejmirtes** [comment]
I rebased it on top of 1.5.x because yeah, this is just a bugfix :)

**Comment by ondrejmirtes** [comment]
Thank you very much! The logic seems too easy but I wasn't able to find any flaw in it.

**Pull request merged**